### DISCUSSION TITLE
Links in labels

### DISCUSSION THREAD

**Discussion by joshbruce** [uswds/uswds#3914] [Q&A]
Hello,

It's been a while since I was active here and I'm glad to see the project is still going and seeming to do well.

I remember a while back there was discussion about links inside checkbox labels (or just form labels in general).

A typical user pattern for things like agreeing that you've read the terms of service, that sort of thing.

I remember it being an accessibility nightmare and don't see it in the current design system.

I'm also not trusting other sources at the moment and was curious what recommendation exists, if any?? :)

Apologies if this is mentioned somewhere in the docs - I didn't see it in the places I considered obvious.

Cheers,
Josh

**Comment by mgifford** [comment]
So you'd be looking for guidance on https://designsystem.digital.gov/components/form/

But perhaps some specific examples from https://designsystem.digital.gov/components/checkbox/ or https://designsystem.digital.gov/components/radio-buttons/

**Comment by ryparker** [comment]
For a link such as terms and conditions, keep the link separate from the checkbox label. That gives the link and checkbox distinct keyboard targets and avoids a click having competing navigation and selection behavior. [MDN's label guidance](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/label#interactive_content) recommends avoiding additional interactive elements inside a label.

For example:

```html
<p id="terms-help">
  <a href="/terms">Read the terms and conditions</a> before agreeing.
</p>
<div class="usa-checkbox">
  <input class="usa-checkbox__input" id="agree" name="agree"
    type="checkbox" aria-describedby="terms-help">
  <label class="usa-checkbox__label" for="agree">
    I agree to the terms and conditions
  </label>
</div>
```

The label still activates the checkbox. The link remains independently accessible, and `aria-describedby` associates the supporting text with the checkbox.
